### ISSUE TITLE
Ops used in torch.distributions to lower

### ISSUE THREAD

**Issue by Chillee** [facebookresearch/functorch#47]
HalfCauchy:

`{'div', 'pow', 'sum', 'log', 'lt', 'exp', 'add', 'mul', 'index_put_', 'neg', 'unsqueeze', 'expand', 'sub', '_local_scalar_dense'}`

LKJCholesky:

`{'ge', 'softplus_backward', 'sum', 'tanh_backward', 'softplus', 'sub', 'tanh', 'add', 'mul', 'neg', '_s_where', 'clamp', 'logical_and', 'le', 'expand', 'index_put_'}`

MultivariateNormal:

`{'t', 'getitem', 'view', 'pow', 'sum', 'triangular_solve', 'permute', 'add', 'mul', 'sub'}`

Uniform:

`{'div', 'sigmoid', 'sum', 'softplus_backward', 'softplus', 'copy_', 'add', 'mul', 'neg', 'clamp', 'le', 'gt', 'sub'}`

Bernoulli:

`{'neg', 'binary_cross_entropy_with_logits', 'sub', 'sum'}`

Beta:

`{'sigmoid_backward', 'sigmoid', 'unbind', 'ge', 'sum', 'softplus', 'mul', 'stack', 'sub', 'log', 'softplus_backward', 'add', 'rsub', 'neg', 'logical_and', 'div', 'clamp', 'le', '_s_where'}`

Dirichlet:

`{'sigmoid_backward', 'sigmoid', 'log_sigmoid_backward', 'expand', 'ge', 'slice', 'getitem', 'sum', 'mul', 'constant_pad_nd', 'log', 'sub', 'add', 'rsub', 'neg', 'logical_and', 'log_sigmoid_forward', 'div', 'cumprod', 'copy_', 'clamp', 'le', '_s_where'}`
